examples/RenderManagerD3DCAPIExample.cpp

```diff
@@ -408,107 +422,129 @@ int main(int argc, char* argv[]) {
-			        return 202;
-            }
-          }
-          if ((OSVR_RETURN_SUCCESS != osvrRenderManagerFinishPresentRenderBuffers(
-            render, presentState, renderParams, false))) {
-            std::cerr << "Could not finish presenting render buffers" << std::endl;
-			      osvrDestroyRenderManager(render);
-			      return 203;
-          }
-        } else {
-          // send a black screen.
-          OSVR_RGB_FLOAT black;
-          black.r = black.g = black.b = 0;
-          osvrRenderManagerPresentSolidColorf(render, black);
-        }
-
-        // Timing information
-        if (elapsed_sec.count() >= 2) {
-            std::chrono::duration<double, std::micro> elapsed_usec =
-                end - start;
-            double usec = elapsed_usec.count();
-            std::cout << "Rendering at " << count / (usec * 1e-6) << " fps"
-                      << std::endl;
-            start = end;
-            count = 0;
-        }
-        count++;
-    }
-
+	while (!quit) {
+		size_t renderBufferIndex = frameIndex % 2;
+		frameIndex++;
+
+		// Update the context so we get our callbacks called and
+		// update tracker state.
+		context.update();
+
+		if ((OSVR_RETURN_SUCCESS != osvrRenderManagerGetRenderInfoCollection(
+			render, renderParams, &renderInfoCollection))) {
+			std::cerr << "Could not get render info" << std::endl;
+			osvrDestroyRenderManager(render);
+			return 105;
+		}
+		osvrRenderManagerGetNumRenderInfoInCollection(renderInfoCollection, &numRenderInfo);
+
+
+		renderInfo.clear();
+		for (OSVR_RenderInfoCount i = 0; i < numRenderInfo; i++) {
+			OSVR_RenderInfoD3D11 info;
+			if ((OSVR_RETURN_SUCCESS != osvrRenderManagerGetRenderInfoFromCollectionD3D11(
+				renderInfoCollection, i, &info))) {
+				std::cerr << "Could not get render info " << i
+					<< std::endl;
+				osvrDestroyRenderManager(render);
+				return 106;
+			}
+			renderInfo.push_back(info);
+		}
+		osvrRenderManagerReleaseRenderInfoCollection(renderInfoCollection);
+
+		// Set up to render to the textures for this eye
+		auto renderTargetView = renderBuffers[renderBufferIndex].colorBufferView;
+		auto depthStencilView = depthStencilViews[renderBufferIndex];
+		auto context = renderInfo[0].library.context;
+		context->OMSetRenderTargets(1, &renderTargetView, depthStencilView);
+
+		// Set up the viewport we're going to draw into.
+		//CD3D11_VIEWPORT viewport(static_cast<float>(renderInfo.viewport.left),
+		//                         static_cast<float>(renderInfo.viewport.lower),
+		//                         static_cast<float>(renderInfo.viewport.width),
+		//                         static_cast<float>(renderInfo.viewport.height));
+		CD3D11_VIEWPORT viewport(
+			0.0f,
+			0.0f,
+			static_cast<float>(width),
+			static_cast<float>(height));
+
+		context->RSSetViewports(1, &viewport);
+
+		// Make a grey background
+		FLOAT colorRgba[4] = { 0.3f, 0.3f, 0.3f, 1.0f };
+		context->ClearRenderTargetView(renderTargetView, colorRgba);
```

Comment:
If there are two windows for the two eyes, we're going to have to clear both of them here, not just one of them.